tests/test_utilities.py

```diff
@@ -312,3 +353,150 @@ def get_when_done_or_timeout(
         sleep(1)
 
     raise TimeoutError()
+
+
+TBaseModel = TypeVar("TBaseModel", bound=DefaultBaseModel)
+
+
+class _SchematicGenerationResultDocument(TypedDict, total=False):
```

Comment:
Not sure why we need the whole shabang with `DocumentDatabase` and models here. Why not just `json.dump/load` to a dict?

Reply:
Instead start to manage inside the transient/flush all over again (I don't want every call for generation to start json.dump/load, its just takes more time, and with this implementation its more clean and easier. I actually started with json.dump/load and moved to this once I understand that I dont want to load it everytime..